File: core/src/main/java/com/tvsc/core/model/Serial.java
package com.tvsc.core.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.squareup.moshi.Json;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

/**
 * @author Taras Zubrei
 */
@Data
@NoArgsConstructor
public class Serial {
    private Long id;
    @JsonProperty("seriesName")
    @Json(name = "seriesName")
    private String name;
    private String banner;
    private String overview;
    private LocalDate firstAired;
    private String network;
    private String imdbId;
    private String zap2itId;
    private String airsDayOfWeek;
    private String airsTime;
    private List<String> genre;
    private String rating;
    private String runtime;
    private String status;
    private String poster;
    private List<Season> seasons;
}

File: core/src/main/java/com/tvsc/core/model/Serial.kt
package com.tvsc.core.model

import com.fasterxml.jackson.annotation.JsonProperty

import java.time.LocalDate

/**
 * @author Taras Zubrei
 */
data class Serial (
        @JsonProperty                   var id: Long? = null,
        @JsonProperty("seriesName")     var name: String? = null,
        @JsonProperty                   var banner: String? = null,
        @JsonProperty                   var overview: String? = null,
        @JsonProperty                   var firstAired: LocalDate? = null,
        @JsonProperty                   var network: String? = null,
        @JsonProperty                   var imdbId: String? = null,
        @JsonProperty                   var zap2itId: String? = null,
        @JsonProperty                   var airsDayOfWeek: String? = null,
        @JsonProperty                   var airsTime: String? = null,
        @JsonProperty                   var genre: List<String>? = null,
        @JsonProperty                   var rating: String? = null,
        @JsonProperty                   var runtime: String? = null,
        @JsonProperty                   var status: String? = null,
                                        var poster: String? = null,
                                        var seasons: List<Season>? = null
    //    String actors;
    //    String contentRating;

)
